metalearners/grid_search.py

```diff
@@ -312,7 +346,22 @@ def fit(
                     )
                 )
 
-        parallel = Parallel(n_jobs=self.n_jobs, verbose=self.verbose)
-        raw_results = parallel(delayed(_fit_and_score)(job) for job in jobs)
-        self.raw_results_ = raw_results
-        self.results_ = _format_results(results=raw_results)
+        self.grid_size_ = len(jobs)
+        self.raw_results_: list[GSResult] | Generator[GSResult, None, None] | None
+        self.results_: pd.DataFrame | None
+
+        return_as = "list" if self.store_raw_results else "generator_unordered"
+        parallel = Parallel(
+            n_jobs=self.n_jobs, verbose=self.verbose, return_as=return_as
+        )
+        self.raw_results_ = parallel(delayed(_fit_and_score)(job) for job in jobs)
+        if self.store_results:
+            self.results_ = _format_results(results=self.raw_results_)  # type: ignore
+            if not self.store_raw_results:
+                # This just checks that the generator is empty
```

Comment:
I'm not sure I get what we want to do if the generator is not empty. :/

Reply:
The generator should never be not empty, it should be finished as it should be used by `_format_results`. I wrote it just to check anything unexpected happened while testing. I think we can remove it.
[e34b751](https://github.com/Quantco/metalearners/pull/62/commits/e34b7512c9c8f2e0799d3e77522c257a4a82fabb)